tests/integration/suite/sentry/sentry.go

```diff
@@ -0,0 +1,406 @@
+			require.Contains(t, grpcStatus.Message(), "not enabled")
+		})
+
+		t.Run("issue a certificate with insecure validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			res, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:                        defaultAppID,
+				Namespace:                 defaultNamespace,
+				CertificateSigningRequest: defaultAppCSR,
+				TokenValidator:            sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.NoError(t, err)
+			require.NotEmpty(t, res.WorkloadCertificate)
+		})
+
+		t.Run("insecure validator is the default", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			res, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:                        defaultAppID,
+				Namespace:                 defaultNamespace,
+				CertificateSigningRequest: defaultAppCSR,
+				// TokenValidator:            sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.NoError(t, err)
+			require.NotEmpty(t, res.WorkloadCertificate)
+		})
+
+		t.Run("fails with missing CSR", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:        defaultAppID,
+				Namespace: defaultNamespace,
+				// CertificateSigningRequest: defaultAppCSR,
+				TokenValidator: sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.PermissionDenied, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "CSR is required")
+		})
+
+		t.Run("fails with missing namespace", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id: defaultAppID,
+				// Namespace:                 defaultNamespace,
+				CertificateSigningRequest: defaultAppCSR,
+				TokenValidator:            sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.PermissionDenied, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "namespace is required")
+		})
+
+		t.Run("fails with invalid CSR", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:                        defaultAppID,
+				Namespace:                 defaultNamespace,
+				CertificateSigningRequest: []byte("-----BEGIN PRIVATE KEY-----\nMC4CAQAwBQYDK2VwBCIEIAYIsKL0xkTkAXDhUN6eDheqODEOGyFZ04jsgFNCFxZf\n-----END PRIVATE KEY-----"),
```

Comment:
Please don't do this, again re committing private keys is bad. Some random garbage data here is fine.

Reply:
Done